commands/project_doctor.go

```diff
@@ -0,0 +1,213 @@
+	if err := cmd.AnalyzeConditionList(compendium); err != nil {
+		// Directly returning the framework error to skip the expanded help.
+		// A failing state is self-descriptive.
+		return cli.NewExitError(fmt.Sprintf("%v", err), 1)
+	}
+
+	return nil
+}
+
+// RunCompendium lists all conditions to be checked in the analysis.
+func (cmd *ProjectDoctor) RunCompendium(ctx *cli.Context) error {
+	compendium, _ := cmd.GetConditionCollection()
+	cmd.out.Info("There are %d conditions in the repertoire.", len(compendium))
+	fmt.Println(compendium)
+
+	return nil
+}
+
+// AnalyzeConditionList checks each registered condition against environment state.
+func (cmd *ProjectDoctor) AnalyzeConditionList(conditions ConditionCollection) error {
+	var returnVal error
+
+	failing := ConditionCollection{}
+	for _, condition := range conditions {
+		cmd.out.Spin(fmt.Sprintf("Examining project environment for %s", condition.Name))
+		if found := cmd.Analyze(condition); !found {
+			cmd.out.Info("Not Affected by: %s [%s]", condition.Name, condition.Id)
+		} else {
+			switch condition.Severity {
+			case ConditionSeverityWARNING:
+				cmd.out.Warning("Condition Detected: %s [%s]", condition.Name, condition.Id)
+				failing[condition.Id] = condition
+				break
+			case ConditionSeverityERROR:
+				cmd.out.Error("Condition Detected: %s [%s]", condition.Name, condition.Id)
+				failing[condition.Id] = condition
+				if returnVal == nil {
+					returnVal = errors.New("Diagnosis found at least one failing condition.")
+				}
+				break
+			default:
+				cmd.out.Info("Condition Detected: %s [%s]", condition.Name, condition.Id)
+			}
+		}
+	}
+
+	if len(failing) > 0 {
+		color.Red("\nThere were %d problems identified out of %d checked.\n", len(failing), len(conditions))
+		fmt.Println(failing)
+	}
+
+	return returnVal
+}
+
+// GetConditionCollection assembles a list of all conditions.
+func (cmd *ProjectDoctor) GetConditionCollection() (ConditionCollection, error) {
+	conditions := cmd.Config.Doctor
+
+	// @TODO move these to outrigger.yml once we have pure shell facilities.
+	eval := ProjectEval{cmd.out, cmd.Config}
+	sync := ProjectSync{}
+	syncName := sync.GetVolumeName(cmd.Config, eval.GetWorkingDirectory())
+
+	// @todo we should have a way to determine if the project wants to use sync.
+	item1 := &Condition{
+		Id:           "sync-container-not-running",
+		Name:         "Sync Container Not Working",
+		Test:         []string{fmt.Sprintf("$(id=$(docker container ps -aq --filter 'name=^/%s$'); docker top $id &>/dev/null)", syncName)},
+		Diagnosis:    "The Sync container for this project is not available.",
+		Prescription: "Run 'rig project sync:start' before beginning work. This command may be included in other project-specific tasks.",
+		Severity:     ConditionSeverityWARNING,
+	}
+	if _, ok := conditions["sync-container-not-running"]; !ok {
+		conditions["sync-container-not-running"] = item1
+	}
+
+	item2 := &Condition{
```

Comment:
Should we also add a test to the effect of `ps aux | grep unison | $syncName` to check for the process on the local machine?